wgpu-core/src/device/mod.rs

```diff
@@ -1357,18 +1363,24 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         }
 
         // TODO: push constants
+        let mut count_validator = binding_model::BindingTypeMaxCountValidator::default();
         let pipeline_layout = {
             let (bind_group_layout_guard, _) = hub.bind_group_layouts.read(&mut token);
-            let descriptor_set_layouts = bind_group_layout_ids
-                .iter()
-                .map(|&id| &bind_group_layout_guard[id].raw);
+            let descriptor_set_layouts = bind_group_layout_ids.iter().map(|&id| {
```

Comment:
that's dangerous thing to do: it doesn't iterate, and instead it constructs an iterator and passes it to gfx-hal.
So it implicitly relies on gfx-hal actually iterating the elements, but this isn't expressed in the type contract.

Let's instead just have a nice `for` loop here for collecting the limits